systemvm/debian/opt/cloud/bin/setup/cloud-early-config

```diff
@@ -71,16 +71,16 @@ config_guest() {
 
 get_boot_params() {
   case $HYPERVISOR in
-     xen-pv)
+     xen-pv|xen-domU)
```

Comment:
@rhtyd you should not add `xen-domU` to this case, because in fully patched Xenserver one HVM instance might be reported as `xen-domU` which in reality is should be `xen-hvm`. Basically this is the output of `virt-what`.
I'll put another comment to describe further.

Reply:
in `hypervisor()` function above, line 44-52 should be moved to the very beginning of the function as the very first step to detect type of Xen, then let `virt-what` does its job. The reason is some HVM instances on Xen still being reported as `xen-domU` (which most probably they are pvhvm or something). But detecting if an instance is on PV or HVM through `dmesg` is accurate.